### PR TITLE
DDCNL-5040

### PR DESCRIPTION
**Pull request by Dandyboyo2**

### CODE REVIEW

File: app/controllers/AuthorisationController.scala [hunk 1]
```diff
@@ -39,7 +38,7 @@ class AuthorisationController @Inject()(
   }
 
   def logout = unauthenticatedAction {
-    implicit request =>
+    //implicit request =>
```
Comment (parkeradam) [inline app/controllers/AuthorisationController.scala:41]: If commented out, should be removed.

File: app/controllers/UpdateRelationshipController.scala [hunk 2]
```diff
@@ -80,24 +79,23 @@ class   UpdateRelationshipController @Inject()(
         case NonFatal(_) => Ok(decisionV(CheckClaimOrCancelDecisionForm.form))
       }
   }
-
+//TODO - Refactor to address warning
```
Comment (parkeradam) [inline app/controllers/UpdateRelationshipController.scala:82]: Has this been done?

File: app/controllers/UpdateRelationshipController.scala [hunk 3]
```diff
@@ -118,7 +116,7 @@ class   UpdateRelationshipController @Inject()(
         case NonFatal(_) => Ok(reasonForChange(MakeChangesDecisionForm.form))
       }
   }
-
+  //TODO - Refactor to address warning
```
Comment (parkeradam) [inline app/controllers/UpdateRelationshipController.scala:119]: Has this been done?

File: app/forms/RegistrationForm.scala [hunk 4]
```diff
@@ -34,11 +34,11 @@ import javax.inject.Inject
 
 class RegistrationForm@Inject()(applicationConfig: ApplicationConfig) {
 
-  private def maxLengthWithError(maxLength: Int, error: String = "error.maxLength"): Constraint[String] = Constraint[String]("constraint.maxLength", maxLength) { o =>
+  private def maxLengthWithError(maxLength: Int, error: String): Constraint[String] = Constraint[String]("constraint.maxLength", maxLength) { o =>
     if (o == null) Invalid(ValidationError(error, maxLength)) else if (o.length <= maxLength) Valid else Invalid(ValidationError(error, maxLength))
   }
 
-  private def nonEmptyTrimmer(error: String = "error.required"): Mapping[String] =
+  private def nonEmptyTrimmer(error: String): Mapping[String] =
```
Comment (parkeradam) [inline app/forms/RegistrationForm.scala:41]: While not an error I am curious about why all these defaults have been removed.

File: app/forms/coc/DivorceSelectYearForm.scala [hunk 5]
```diff
@@ -66,7 +66,7 @@ class DivorceSelectYearForm@Inject()(applicationConfig: ApplicationConfig, timeS
     case _ => Invalid(ValidationError("pages.divorce.date.error.mandatory"))
 
   }
-
+  //TODO - Refactor to address warning
```
Comment (parkeradam) [inline app/forms/coc/DivorceSelectYearForm.scala:69]: Has this been done

File: build.sbt [hunk 6]
```diff
@@ -33,8 +47,10 @@ lazy val microservice: Project = Project(appName, file("."))
...
     majorVersion := 7
   )
-
+scalacOptions ++= Seq("-P:silencer:pathFilters=routes")
+scalacOptions ++= suppressedImports
 scalacOptions ++= Seq(
   "-Xmaxerrs", "1000", // Maximum errors to print
-  "-Xmaxwarns", "1000" // Maximum warnings to print
+  "-Xmaxwarns", "1000", // Maximum warnings to print
+  //"-Xfatal-warnings"
```
Comment (parkeradam) [inline build.sbt:55]: remove commented out line

**Review by parkeradam — CHANGES_REQUESTED** [review]

File: app/controllers/AuthorisationController.scala (hunk 1 above)
Comment (Dandyboyo2) [reply app/controllers/AuthorisationController.scala]: Code amended

File: build.sbt (hunk 6 above)
Comment (Dandyboyo2) [reply build.sbt]: Code amended

File: app/forms/coc/DivorceSelectYearForm.scala (hunk 5 above)
Comment (Dandyboyo2) [reply app/forms/coc/DivorceSelectYearForm.scala:69]: This was added following a discussion with Kurt, will likely result in a separate ticket, I'll raise further discussion with him.

File: app/controllers/UpdateRelationshipController.scala (hunk 3 above)
Comment (Dandyboyo2) [reply app/controllers/UpdateRelationshipController.scala:119]: This was added following a discussion with Kurt, will likely result in a separate ticket, I'll raise further discussion with him.

File: app/controllers/UpdateRelationshipController.scala (hunk 2 above)
Comment (Dandyboyo2) [reply app/controllers/UpdateRelationshipController.scala:82]: This was added following a discussion with Kurt, will likely result in a separate ticket, I'll raise further discussion with him.

File: app/forms/RegistrationForm.scala (hunk 4 above)
Comment (Dandyboyo2) [reply app/forms/RegistrationForm.scala:41]: they give warnings for not being used, the defaults were defined in 2 places

**Review by parkeradam — APPROVED** [review]

**Pull request merged**